adoc/admin-security-certificates.adoc

```diff
@@ -217,38 +268,56 @@ sudo cat /etc/kubernetes/admin.conf
 
 . Log in to the master node and regenerate the certificates:
 +
-* Replace the oidc-dex secret:
-+
-[source,bash]
-----
-cd /etc/kubernetes/pki
-----
+* Replace the oidc-dex server certificate:
 +
-Sign the oidc-dex server certificate with the root CA `ca.crt/ca.key` and SAN
-as <CONTROL_PLANE_IP_ADDRESS/FQDN>. Then update the {kube} cluster secret data
+Sign the oidc-dex server certificate with the CA `ca.crt/ca.key` and SAN
+as origin was. Then update the {kube} cluster secret data
 `ca.crt`, `tls.crt`, and `tls.key` with base64 encoded:
 +
 [source,bash]
 ----
-cd /etc/kubernetes
-sudo kubectl --kubeconfig=admin.conf edit secret oidc-dex-cert -n kube-system
-sudo kubectl --kubeconfig=admin.conf delete pod -lapp=oidc-dex -n kube-system
+sudo mkdir -p /etc/kubernetes/pki.bak
+cd /etc/kubernetes/pki.bak
+sudo kubectl --kubeconfig=/etc/kubernetes/admin.conf get secret oidc-dex-cert -n kube-system -o yaml | sudo tee oidc-dex-cert.yaml > /dev/null
+
+sudo kubectl --kubeconfig=/etc/kubernetes/admin.conf edit secret oidc-dex-cert -n kube-system
+sudo kubectl --kubeconfig=/etc/kubernetes/admin.conf delete pod -lapp=oidc-dex -n kube-system
 ----
 
-* Replace the oidc-gangway secret:
+* Replace the oidc-gangway server certificate:
++
+Sign the oidc-gangway server certificate with the CA `ca.crt/ca.key` and SAN
+as origin was. Then update the {kube} cluster secret data
+`ca.crt`, `tls.crt`, and `tls.key` with base64 encoded.
 +
 [source,bash]
 ----
-cd /etc/kubernetes/pki
+sudo mkdir -p /etc/kubernetes/pki.bak
+cd /etc/kubernetes/pki.bak
+sudo kubectl --kubeconfig=/etc/kubernetes/admin.conf get secret oidc-gangway-cert -n kube-system -o yaml | sudo tee oidc-gangway-cert.yaml > /dev/null
+
+sudo kubectl --kubeconfig=/etc/kubernetes/admin.conf edit secret oidc-gangway-cert -n kube-system
+sudo kubectl --kubeconfig=/etc/kubernetes/admin.conf delete pod -lapp=oidc-gangway -n kube-system
 ----
+
+* Replace the kubelet server certificate:
++
+[IMPORTANT]
+====
+You need to generate kubelet server certificate for all the nodes, and the kubelet server key only exists on the control plane nodes. Therefore, after re-signed kubelet server certificate/key for worker nodes, you have to copy the worker nodes kubelet server certificate/key from control plane node.
```

Comment:
```suggestion
You need to generate kubelet server certificate for all the nodes on one of control plane nodes, because the kubelet CA certificate key only exists on the control plane nodes. Therefore, after generated re-signed kubelet server certificate/key for worker nodes, you have to copy each kubelet server certificate/key from control plane node to the corresponding worker node.
```

Reply:
Appreciated.